BraveRewardsUI/Tipping/TippingViewController.swift

```diff
@@ -72,9 +155,46 @@ class TippingViewController: UIViewController, UIViewControllerTransitioningDele
   }
   
   @objc private func tappedSendTip() {
-    tippingView.setTippingConfirmationVisible(true, animated: true)
-    DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) { [weak self] in
-      self?.dismiss(animated: true)
+    if let selectedIndex = self.tippingView.optionSelectionView.selectedOptionIndex {
+      let amount = self.tippingView.optionSelectionView.options[selectedIndex].value.doubleValue
+      
+      state.ledger.listRecurringTips { [weak self] info in
+        guard let self = self else { return }
+        
+        // If the user has recurring tips, remove the monthly tip
+        if info.firstIndex(where: { $0.id == self.publisherInfo.id }) != nil {
+          self.state.ledger.removeRecurringTip(publisherId: self.publisherInfo.id)
+        }
+        
+        // Add recurring tips if there is none..
+        if self.tippingView.optionSelectionView.isMonthly {
+          self.state.ledger.addRecurringTip(publisherId: self.publisherInfo.id, amount: amount)
```

Comment:
You shouldn't be tipping directly if its a recurring tip, needs a `return` after this line.

Logic from core:

https://github.com/brave/brave-core/blob/150052b89ab409cb0dd933237c9c1395bfc262c7/components/brave_rewards/browser/rewards_service_impl.cc#L2124-L2145